refixed issue #3 and fixed issue #4

Joungkyun opened the pull request:
fixed #3 and #4

ichiriac commented:
the tests does not pass in debug mode : 
https://travis-ci.org/expressif/pecl-event-libevent/builds/104572517

```
php: /home/travis/job-17.2/include/php/Zend/zend_types.h:820: zval_addref_p: Assertion `(((*(pz)).u1.v.type_flags & (1<<2)) != 0)' failed.
002+ Aborted (core dumped)

004+ [Mon Jan 25 08:17:14 2016]  Script:  '/home/travis/build/expressif/pecl-event-libevent/tests/libevent002.php'
005+ /home/travis/build/expressif/pecl-event-libevent/travis/pecl/pecl-libevent/libevent.c(579) :  Freeing 0x2BA1D1A01B40 (56 bytes), script=/home/travis/build/expressif/pecl-event-libevent/tests/libevent002.php
006+ [Mon Jan 25 08:17:14 2016]  Script:  '/home/travis/build/expressif/pecl-event-libevent/tests/libevent002.php'
007+ /home/travis/build/expressif/pecl-event-libevent/travis/pecl/pecl-libevent/libevent.c(365) :  Freeing 0x2BA1D1A79118 (24 bytes), script=/home/travis/build/expressif/pecl-event-libevent/tests/libevent002.php
008+ [Mon Jan 25 08:17:14 2016]  Script:  '/home/travis/build/expressif/pecl-event-libevent/tests/libevent002.php'
009+ /home/travis/build/expressif/pecl-event-libevent/travis/pecl/php-master/Zend/zend_closures.c(371) :  Freeing 0x2BA1D1A7B000 (296 bytes), script=/home/travis/build/expressif/pecl-event-libevent/tests/libevent002.php
010+ === Total 3 memory leaks detected ===

013+ [Mon Jan 25 08:17:14 2016]  Script:  '/home/travis/build/expressif/pecl-event-libevent/tests/libevent003.php'
014+ /home/travis/build/expressif/pecl-event-libevent/travis/pecl/pecl-libevent/libevent.c(969) :  Freeing 0x2B2E1E05F400 (88 bytes), script=/home/travis/build/expressif/pecl-event-libevent/tests/libevent003.php
015+ Last leak repeated 1 time
016+ [Mon Jan 25 08:17:14 2016]  Script:  '/home/travis/build/expressif/pecl-event-libevent/tests/libevent003.php'
017+ /home/travis/build/expressif/pecl-event-libevent/travis/pecl/pecl-libevent/libevent.c(365) :  Freeing 0x2B2E1E079188 (24 bytes), script=/home/travis/build/expressif/pecl-event-libevent/tests/libevent003.php
018+ Last leak repeated 1 time
019+ === Total 4 memory leaks detected ===
```

Joungkyun commented:
my test result is follow. only libevent001.phpt has memory leak but lievent002 and libevent003 is safe. I think that problems are travis-ci environment (PHP 7.1)

[root@an3 z]$ make test PHP_EXECUTABLE=/usr/bin/php

Build complete.
Don't forget to run 'make test'.


=====================================================================
PHP         : /usr/bin/php
PHP_SAPI    : cli
PHP_VERSION : 7.0.2AnNyung
ZEND_VERSION: 3.0.0
PHP_OS      : Linux - Linux an3.pkg.oops.org 3.10.0-327.4.4.el7.x86_64 #1 SMP Tue Jan 5 16:07:00 UTC 2016 x86_64
INI actual  : /root/work/github/z/tmp-php.ini
More .INIs  :
CWD         : /root/work/github/z
Extra dirs  :
VALGRIND    : valgrind-3.10.0
=====================================================================
TIME START 2016-01-26 14:51:26
=====================================================================
LEAK pecl/libevent - general [tests/libevent001.phpt]
PASS pecl/libevent - bug https://github.com/expressif/pecl-event-libevent/issues/2 [tests/libevent002.phpt]
PASS pecl/libevent - bug https://github.com/expressif/pecl-event-libevent/issues/3 [tests/libevent003.phpt]
=====================================================================
TIME END 2016-01-26 14:51:42

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   35
---------------------------------------------------------------------

Number of tests :    3                 3
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests leaked    :    1 ( 33.3%) ( 33.3%)
Tests passed    :    2 ( 66.7%) ( 66.7%)
---------------------------------------------------------------------
Time taken      :   16 seconds
=====================================================================

=====================================================================
LEAKED TEST SUMMARY
---------------------------------------------------------------------
pecl/libevent - general [tests/libevent001.phpt]
=====================================================================

You may have found a problem in PHP.
This report can be automatically sent to the PHP QA team at
http://qa.php.net/reports and http://news.php.net/php.qa.reports
This gives us a better understanding of PHP's behavior.
If you don't want to send the report immediately you can choose
option "s" to save it.  You can then email it to qa-reports@lists.php.net later.
Do you want to send this report now? [Yns]:

Joungkyun commented:
fixed memory leak for event_new, event_buffer_new, event_base_new and fixed various bugs.

The remains leak of travis-ci log is bug nothing to do with bug #3 and my patches.
